CI: Building fails on nightly compiler because of rust-lang/rust#79813

```
   Compiling redis v2.0.0-rc.1 (/home/runner/work/redis-rs/redis-rs/redis)
error: trailing semicolon in macro used in expression position
   --> redis/src/parser.rs:80:5
    |
 80 | /     opaque!(any_send_sync_partial_state(
 81 | |         any()
 82 | |             .then_partial(move |&mut b| {
 83 | |                 if count > MAX_RECURSE_DEPTH {
...   |
323 | |             })
324 | |     ))
    | |______^
    |
    = note: macro invocations at the end of a block are treated as expressions
    = note: to ignore the value produced by the macro, add a semicolon after the invocation of `opaque`
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
    = note: `#[deny(semicolon_in_expressions_from_macros)]` (part of `#[deny(future_incompatible)]`) on by default
    = note: this error originates in the macro `opaque` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `redis` (lib) due to 2 previous errors
make: *** [Makefile:2: build] Error 101
```

--- a/redis/src/parser.rs
+++ b/redis/src/parser.rs
@@ -1,3 +1,8 @@
+// TODO remove this `allow` once `combine` released a fix. Upstream bug: https://github.com/Marwes/combine/issues/372
+#![allow(
+    semicolon_in_expressions_from_macros,
+    reason = "This lint is on in nightly since 2026-07-16, but `combine-4.6.7` violates it in `opaque`. As we cannot decorate directly, we allow it for the whole module for now"
+)]
 use std::{
     io::{self, Read},
     str,

